table/block_based_table_reader.cc

```diff
@@ -2880,10 +2882,14 @@ void BlockBasedTable::MultiGet(const ReadOptions& read_options,
       bool matched = false;  // if such user key matched a key in SST
       bool done = false;
       for (iiter->Seek(key); iiter->Valid() && !done; iiter->Next()) {
-        DataBlockIter biter;
-        NewDataBlockIterator<DataBlockIter>(
-            rep_, read_options, iiter->value(), &biter, false,
-            true /* key_includes_seq */, get_context);
+        if (iiter->value().offset() != bhandle.offset() ||
+            iiter->value().size() != bhandle.size()) {
```

Comment:
I believe offset is enough. We can skip the size check. We already do it when reseeking in iterator since long ago. In this way, actually only offset of the previous block handle needs to be stored.

Reply:
Actually I think 0 is a valid offset. I just got a failure in asan_crash because of it. But I believe we can just check for offset by initializing it to ULONG_MAX.

Reply:
Yes 0 is a valid offset. I agree that max int is a good idea. Make sure you use the one defined in port.h.